server/util/networking/networking.go

```diff
@@ -346,6 +336,35 @@ func (a *HostNetAllocator) unlock(netIdx int) {
 //	# add a route in the root namespace so that traffic to 192.168.0.3 hits 10.0.0.2, the veth0 end of the pair
 //	$ sudo ip route add 192.168.0.3 via 10.0.0.2
 func SetupVethPair(ctx context.Context, netNamespace, vmIP string, vmIdx int) (_ func(context.Context) error, err error) {
+	// Keep a list of cleanup work to be done. For each resource we create in
+	// this func that needs to be explicitly cleaned up, we append a cleanup
+	// task to this list. Cleanup work is done in the reverse order in which
+	// it's added to this list (i.e., it's a stack).
+	var cleanupStack []func(ctx context.Context) error
```

Comment:
I find this logic a bit complex to follow. For example it takes some mental work to figure out the order the cleanup tasks are executing (i.e. "if I want to cleanup the route before I unlock the net idx, when should I add it to the slice?") I would personally find it a bit more readable to use booleans

Ex. 
if shouldCleanupRoute {}
if shouldCleanupRule {}